docs/corelang.md

```diff
@@ -0,0 +1,1025 @@
+  `{a: 1, b: 2} + {b: 3, c: 4} --> {"a": 1, "b": 3, "c": 4}`.
+- Adding anything else yields an error.
+
+The filter `$x - $y` subtracts `$y` from `$x` as follows:
+
+- Subtracting numbers yields their difference, similar to addition.
+- Subtracting arrays yields the
+  left array with all elements contained in the right array removed; i.e.
+  `[1, 2, 3, 4] - [2, 4] --> [1, 3]`.
+- Subtracting anything else yields an error.
+
+{#mul-div}
+
+### Multiplication / division
+
+The filter `$x * $y` multiplies two values as follows:
+
+- Multiplying numbers yields their product, similar to addition.
+- Multiplying a string with an integer `$n` yields
+  the `$n`-fold concatenation of the string, i.e.
+  `"abc" * 3 --> "abcabcabc"`.
+  If `$n <= 0`, then this yields `null`, i.e.
+  `0 * "abc" --> null`.
+- Multiplying two objects merges them recursively.
+  In particular,
+  `$x * {k: v, ...}` yields
+  `($x + {k: $x[k] * v}) * {...}` if both `v` and `$x[k]` are objects, else
+  `($x + {k: v}) * {...}`.
+  For example,
+  `{a: {b: 0, c: 2}, e: 4} * {a: {b: 1, d: 3}, f: 5} -->
+  {"a": {"b": 1, "c": 2, "d": 3}, "e": 4, "f": 5}`.
+- Multiplying anything else yields an error.
+
+The filter `$x / $y` divides two values as follows:
+
+- Dividing a number by a number yields their quotient as floating-point number.
+  To perform this operation, both arguments are first converted to floating-point numbers.
+  For example, `1 / 2 --> 0.5`.
+- Dividing a string by a string *splits* `$x` by `$y`, yielding an array of strings.
+  For example,
+  `"foobarfoobazfoo" / "foo" --> ["", "bar", "baz", ""]`.
+  If `$y` is empty, then `$x / $y` yields an array with each character of the input as separate string.
+  For example,
+  `"🧑‍🔬 is 🤔" / "" --> ["🧑","‍","🔬"," ","i","s"," ","🤔"]`.
+
+::: Advanced
+
+You can round-trip string division with `join($y)`.
+For example:
+
+- `"foobarfoobazfoo" / "foo" | join("foo") --> "foobarfoobazfoo"`
+- `"🧑‍🔬 is 🤔" / "" | join("") --> "🧑‍🔬 is 🤔"`
+
+:::
+
+::: Compatibility
+
+In jq, division by 0 yields an error, whereas
+in jaq, `n / 0` yields
+`nan` if `n == 0`,
+` infinite` if `n > 0`, and
+`-infinite` if `n < 0`.
+jaq's behaviour is closer to the IEEE standard for floating-point arithmetic (IEEE 754).
+
+:::
+
+### Modulus
+
+The filter `$x % $y` calculates the modulus of two numbers,
+and fails for anything else.
+For example,
+`5 % 2 --> 1`.
+Any of the two numbers can also be a floating-point number;
+however, the result of this may be unexpected.
+For example,
+`5.1 % 2 --> 1.0999999999999996` and
+`5.5 % 2 --> 1.5`.
+
+
+
+## Keywords
+
+This section lists all filters that start with a reserved keyword.
+
+### if-then-else
+
+The filter `if p then f else g end` runs the filter `p` with its input.
+For every output of `p`, if its [boolean value](#booleans) is `true`,
+the outputs of `f` run with the original input are returned, else
+the outputs of `g` run with the original input are returned.
```

Comment:
Can't find that optional `else` is documented anywhere. Impressive and nice work with the documentation btw, i've started to skim thru it